stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/message/experimental/MessageSendingService.kt

```diff
@@ -0,0 +1,157 @@
+package io.getstream.chat.android.offline.message.experimental
+
+import io.getstream.chat.android.client.errors.ChatError
+import io.getstream.chat.android.client.extensions.enrichWithCid
+import io.getstream.chat.android.client.extensions.uploadId
+import io.getstream.chat.android.client.logger.ChatLogger
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.utils.Result
+import io.getstream.chat.android.client.utils.SyncStatus
+import io.getstream.chat.android.client.utils.flatMapSuspend
+import io.getstream.chat.android.client.utils.onSuccess
+import io.getstream.chat.android.client.utils.onSuccessSuspend
+import io.getstream.chat.android.core.ExperimentalStreamChatApi
+import io.getstream.chat.android.offline.experimental.channel.logic.ChannelLogic
+import io.getstream.chat.android.offline.experimental.channel.state.ChannelState
+import io.getstream.chat.android.offline.experimental.global.GlobalState
+import io.getstream.chat.android.offline.experimental.plugin.logic.LogicRegistry
+import io.getstream.chat.android.offline.message.attachment.UploadAttachmentsWorker
+import io.getstream.chat.android.offline.message.attachment.generateUploadId
+import io.getstream.chat.android.offline.message.getMessageType
+import io.getstream.chat.android.offline.message.hasPendingAttachments
+import io.getstream.chat.android.offline.message.isEphemeral
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.filterNot
+import kotlinx.coroutines.launch
+import java.util.Date
+import java.util.UUID
+
+@ExperimentalStreamChatApi
+internal class MessageSendingService(
+    private val logicRegistry: LogicRegistry,
+    // TODO: Pass StateRegistry instead of channelState and globalState once ChatDomain is removed.
+    private val channelState: ChannelState,
+    private val globalState: GlobalState,
+    private val uploadAttachmentsWorker: UploadAttachmentsWorker,
+    private val scope: CoroutineScope,
+) {
+    private val logger = ChatLogger.get("MessageSendingService")
+    private var jobsMap: Map<String, Job> = emptyMap()
+
+    private val channelLogic: ChannelLogic by lazy {
```

Comment:
Please don't instantiate dependencies here, this can be done in `MessageSendingServiceFactory`.